### PR TITLE
New branch with star rating functionality for related products (fixed package-lock.json error)

### PR DESCRIPTION
**Pull request by jabrake**
- Passing down average rating data to relevant components
- Created new helper to fetch ratings data from API for related products
- Updated a helper function to account for edge case when there are no ratings for a product
- Added Tatt's helper to calculate average rating to component that needs to do calculation
- Added Tatt's HTML to render ratings as stars in card components
- Removed weird package-lock.json error from previous branch/commits

@alizehrehman @tchitrakorn @matthewwrobel  this is ready for review! Thanks Tatt for your star rating functionality, worked like a charm. It's getting late so I tagged all of you depending on who may get to it first 😄

### CODE REVIEW

File: client/related_items/ItemCard.jsx [hunk 1]
```diff
@@ -46,6 +47,15 @@ class ItemCard extends React.Component {
             });
           });
       });
+
+    getProductRatings(this.props.id)
+      .then(ratingsMeta => {
+        let averageRating = computeAverageRating(ratingsMeta.ratings)[1];
```
Comment (tchitrakorn) [inline client/related_items/ItemCard.jsx:53]: Please let me know if the [1] indexing is not very readable! I plan to refactor a lot of my code next week and readability is definitely going to be my top-priority. I can change the helper to return an object instead for a more readable access.

File: client/related_items/ItemCard.jsx [hunk 2]
```diff
@@ -64,7 +74,7 @@ class ItemCard extends React.Component {
           <p id='rp-card-category'>{this.state.category}</p>
           <p id='rp-card-name'>{this.state.name}</p>
           <p id='rp-card-price'>${this.state.price}</p>
-          <p id='rp-card-rating'>{this.state.rating}</p>
+          <span class='stars' style={{'--rating': this.state.rating}}></span>
```
Comment (tchitrakorn) [inline client/related_items/ItemCard.jsx:77]: I hope this works!

**Review by tchitrakorn — APPROVED** [review]
Hi @jabrake, everything looks good! Love your code, very neat!

File: client/related_items/ItemCard.jsx (hunk 1 above)
Comment (jabrake) [reply client/related_items/ItemCard.jsx:53]: Haha I just had to look at your function and understood right away. Agreed that eventually an object would make this more readable - let me know if/when you do that so I can update how I'm using the function on my end as well!

File: client/related_items/ItemCard.jsx (hunk 2 above)
Comment (jabrake) [reply client/related_items/ItemCard.jsx:77]: It does!

**Pull request merged**